internal/engine/worker/contract/lido/worker.go

```diff
@@ -0,0 +1,545 @@
+			lido.EventHashStakedETHWithdrawalNFTWithdrawalRequested,
+			lido.EventHashStakedETHWithdrawalNFTWithdrawalClaimed,
+			lido.EventHashStakedMATICSubmitEvent,
+			lido.EventHashStakedMATICRequestWithdrawEvent,
+			lido.EventHashStakedMATICClaimTokensEvent,
+			lido.EventHashStakedETHTransferShares,
+		},
+	}
+}
+
+func (w *worker) Match(_ context.Context, task engine.Task) (bool, error) {
+	return task.GetNetwork().Source() == filter.NetworkEthereumSource, nil
+}
+
+// Transform Ethereum task to feed.
+func (w *worker) Transform(ctx context.Context, task engine.Task) (*schema.Feed, error) {
+	ethereumTask, ok := task.(*source.Task)
+	if !ok {
+		return nil, fmt.Errorf("invalid task type: %T", task)
+	}
+
+	// Build default lido feed from task.
+	feed, err := ethereumTask.BuildFeed(schema.WithFeedPlatform(filter.PlatformLido))
+	if err != nil {
+		return nil, fmt.Errorf("build feed: %w", err)
+	}
+
+	// Match and handle ethereum logs.
+	for _, log := range ethereumTask.Receipt.Logs {
+		var (
+			actions []*schema.Action
+			err     error
+		)
+
+		// Match lido core contract events
+		switch {
+		case w.matchStakedETHSubmittedLog(ethereumTask, log):
+			// Add ETH liquidity
+			feed.Type = filter.TypeExchangeLiquidity
+			actions, err = w.transformStakedETHSubmittedLog(ctx, ethereumTask, log)
+		case w.matchStakedETHWithdrawalNFTWithdrawalRequestedLog(ethereumTask, log):
+			// Mint ETH withdrawal NFT
+			feed.Type = filter.TypeExchangeLiquidity
+			actions, err = w.transformStakedETHWithdrawalNFTWithdrawalRequestedLog(ctx, ethereumTask, log)
+		case w.matchStakedETHWithdrawalNFTWithdrawalClaimedLog(ethereumTask, log):
+			// Remove ETH liquidity
+			feed.Type = filter.TypeCollectibleBurn
+			actions, err = w.transformStakedETHWithdrawalNFTWithdrawalClaimedLog(ctx, ethereumTask, log)
+		case w.matchStakedMATICSubmitEventLog(ethereumTask, log):
+			// Add MATIC liquidity
+			feed.Type = filter.TypeExchangeLiquidity
+			actions, err = w.transformStakedMATICSubmitEventLog(ctx, ethereumTask, log)
+		case w.matchStakedMATICRequestWithdrawEventLog(ethereumTask, log):
+			// Mint MATIC withdrawal NFT
+			feed.Type = filter.TypeCollectibleMint
+			actions, err = w.transformStakedMATICRequestWithdrawEventLog(ctx, ethereumTask, log)
+		case w.matchStakedMATICClaimTokensEventLog(ethereumTask, log):
+			// Remove MATIC liquidity
+			feed.Type = filter.TypeCollectibleBurn
+			actions, err = w.transformStakedMATICClaimTokensEventLog(ctx, ethereumTask, log)
+		case w.matchStakedETHTransferSharesLog(ethereumTask, log):
+			// Wrap or unwrap wstETH
+			feed.Type = filter.TypeExchangeSwap
+			actions, err = w.transformStakedETHTransferSharesLog(ctx, ethereumTask, log)
+		default:
+			continue
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		feed.Actions = append(feed.Actions, actions...)
+	}
+
+	return feed, nil
+}
+
+// NewWorker creates a new Lido worker.
+func NewWorker(config *config.Module) (engine.Worker, error) {
```

Comment:
1. `NewWorker` should be the last function in `worker.go`
2. Should add more comments of match/transform function